CMSgov/qpp-submissions-docs
# QPPAUTH-2104 Modernize OAuth sample application

Rewrite the OAuth sample app using the latest technologies and conventions, and improve the clarity of the code and documentation.

## Description

- Upgrade target Node.js from v10 -> v18
- Rewrite documentation
- Tokens are cached client-side rather than server-side
- New minimal UI that only an engineer could love
- Increased emphasis on using the ID token on the client side, since the access token is meant for resource servers

## Motivation and Context

This is required both for [QPPPM-2183](https://jira.cms.gov/browse/QPPPM-2183) and the effort around documentation for the upcoming CQR platform, where Auth will likely recommend taking an OAuth-centric approach.


For the Auth ticket, see https://jira.cms.gov/browse/QPPAUTH-2104

## How Has This Been Tested?



This has been tested and verified locally by two Auth engineers, @0xSiO and @daryll.
Testing occurred as follows:
- Configure the application's `.env` file for an OIDC application in CMS IDM (both a `web` and a `native` app were tested)
- Run the application and verify the behavior of each endpoint

## Screenshots (if appropriate):
Contact Luc Street on CMS Slack for a detailed testing report that includes screenshots.

## Types of Changes


## Checklist:


- [ ] I have updated the Last-Updated value on all documentation pages where I changed content.
- [x] My code follows the code style of this project. (As far as I'm aware - let me know if anything doesn't conform)
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.
